test/e2e/db_users_oidc_test.go

```diff
@@ -38,85 +64,84 @@ var _ = Describe("Operator to run db-user with the OIDC feature flags", Ordered,
-			Eventually(k8s.CreateNamespace(testData.Context, testData.K8SClient, config.DefaultOperatorNS)).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).Should(Succeed())
-			k8s.CreateDefaultSecret(testData.Context, testData.K8SClient, config.DefaultOperatorGlobalKey, config.DefaultOperatorNS)
-
-			mgr, err := k8s.BuildManager(&k8s.Config{
-				GlobalAPISecret: client.ObjectKey{
-					Namespace: config.DefaultOperatorNS,
-					Name:      config.DefaultOperatorGlobalKey,
+		By("Creating database users resource", func() {
+			deploy.CreateUsers(testData)
+		})
+		By("Creating a FedAuth resource and verify it is ready", func() {
+			fedAuth := &akov2.AtlasFederatedAuth{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%s-fedauth", testData.Project.Name),
+					Namespace: testData.Resources.Namespace,
 				},
-				WatchedNamespaces: map[string]bool{config.DefaultOperatorNS: true},
-				FeatureFlags:      featureflags.NewFeatureFlags(func() []string { return []string{} }),
-			})
-			Expect(err).NotTo(HaveOccurred())
-
-			go func(ctx context.Context) context.Context {
-				err := mgr.Start(ctx)
-				Expect(err).NotTo(HaveOccurred())
-				return ctx
-			}(testData.Context)
+				Spec: akov2.AtlasFederatedAuthSpec{
+					Enabled:                     true,
+					DataAccessIdentityProviders: &[]string{"abc"},
+					ConnectionSecretRef: common.ResourceRefNamespaced{
+						Namespace: testData.Resources.Namespace,
+						Name:      config.DefaultOperatorGlobalKey,
+					},
+				},
+			}
+			Expect(testData.K8SClient.Create(testData.Context, fedAuth)).Should(Succeed())
+			Eventually(func(g Gomega) {
+				currentFedAuth := &akov2.AtlasFederatedAuth{}
+				g.Expect(testData.K8SClient.Get(context.Background(),
+					types.NamespacedName{
+						Name:      fedAuth.Name,
+						Namespace: fedAuth.Namespace,
+					}, currentFedAuth)).NotTo(HaveOccurred())
+				for _, condition := range currentFedAuth.Status.Conditions {
+					if condition.Type == api.ReadyType {
+						g.Expect(condition.Status).Should(Equal(corev1.ConditionTrue))
+					}
+				}
+			}).WithTimeout(1 * time.Minute).WithPolling(20 * time.Second).Should(Succeed())
 		})
-		By("Creating project and database users resources", func() {
-			deploy.CreateProject(testData)
-			deploy.CreateUsers(testData)
+		By("Try to enabled the OIDC Group feature for the user", func() {
+			currentUser := &akov2.AtlasDatabaseUser{}
+			Expect(testData.K8SClient.Get(context.Background(),
+				types.NamespacedName{
+					Name:      testData.Users[0].Name,
+					Namespace: testData.Users[0].Namespace,
+				}, currentUser)).NotTo(HaveOccurred())
 
-			Eventually(func(g Gomega) bool {
-				for i := range testData.Users {
-					dbUser := testData.Users[i]
+			currentUser.Spec.OIDCAuthType = "IDP_GROUP"
+			currentUser.Spec.PasswordSecret = nil
+			Expect(testData.K8SClient.Update(context.Background(), currentUser)).NotTo(HaveOccurred())
```

Comment:
this needs retry on conflict handling, see https://github.com/mongodb/mongodb-atlas-kubernetes/blob/30b82ed8c680ce5aa616489ca6671dd78b74331e/test/e2e/independent_customroles_test.go#L139-L142

Reply:
Added to both the updates